### PR TITLE
Omit `service-worker-index.html` when `service-worker.js` does not exist

### PR DESCRIPTION
**Pull request by leader22**
### Before submitting the PR, please make sure you do the following
- [x] It's really useful if your PR relates to an outstanding issue, so please reference it in your PR, or create an explanatory one for discussion. In many cases features are absent for a reason.
- [x] This message body should clearly illustrate what problems it solves. If there are related issues, remember to reference them.
- [x] Ideally, include a test that fails without this PR but passes with it. PRs will only be merged once they pass CI. (Remember to `npm run lint`!)
### Tests
-  [x] Run the tests tests with `npm test` or `yarn test`)

----

The `service-worker-index.html` should be generated only when we use ServiceWorker.

### CODE REVIEW

File: test/apps/export-without-serviceworker/test.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+		assert.ok(client_assets.length > 0);
+
+		assert.deepEqual(non_client_assets.sort(), [
+			'global.css',
+			'index.html',
+		].sort());
+	});
+
+	// TODO test timeout, basepath
```
Comment (benmccann) [inline test/apps/export-without-serviceworker/test.ts:28]: it looks like this was carried over from `test/apps/export/test.ts` and isn't needed here

**Comment by leader22** [comment]
@benmccann 
Thank you for the review 😉 
I deleted that comment.

**Comment by benmccann** [comment]
The `service-worker-index.html` seems to be an undocumented feature...

[The template suggests that Sapper generates the file](https://github.com/sveltejs/sapper-template/blob/b51e2f2d9769b4a9d2b9e63c9177ce587bee7a9a/src/service-worker.js#L51), but I'm not sure that it does.

**Comment by benmccann** [comment]
@leader22 would you be able to rebase this PR?

**Comment by leader22** [comment]
@benmccann I resolved conflicts but one test(unrelated to this PR) failed.

```
135 passing (1m)
1 failing

1) export-webpack
     "before all" hook: build app for "injects <link rel=preload> tags":
   Error: Timeout of 10000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves. (/Users/runner/work/sapper/sapper/test/apps/export-webpack/test.ts)
```

Can you re-run that action?

**Pull request merged**